src/gridcoin/gridcoin.cpp

```diff
@@ -500,3 +500,54 @@ void GRC::ScheduleBackgroundJobs(CScheduler& scheduler)
     ScheduleUpdateChecks(scheduler);
     ScheduleBeaconDBPassivation(scheduler);
 }
+
+bool GRC::CleanConfig() {
+    fsbridge::ifstream orig_config(GetConfigFile());
+    if (!orig_config.good()) {
+        return false;
+    }
+
+    bool commit{false};
+    std::vector<std::string> new_config_lines;
+
+    std::string obsolete_keys[] = {"privatekey", "publickey", "PrimaryCPID", "NEURAL_", "enablespeech",
+                                   "suppressupgrade", "UpdatingLeaderboard"};
```

Comment:
Found some more when looking at my old config files. Also probably want to organize this in some way. Maybe alphabetical?
```suggestion
    std::string obsolete_keys[] = {"AutoUpgrade", "cpumining", "enablespeech", "NEURAL_", 
                                   "privatekey", "publickey", "PrimaryCPID", "SessionGuid", 
                                   "suppressupgrade", "tickers", "UpdatingLeaderboard"};
```

Reply:
They can't be fully sorted, as the first key is used to determine if the file is going to be overwritten. I will sort the rest alphabetically. Thanks for the keys.